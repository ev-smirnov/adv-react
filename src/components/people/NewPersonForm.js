import React, { Component } from 'react';
import { reduxForm, Field } from 'redux-form';
import emailValidator from 'email-validator';
import ErrorField from '../common/ErrorField';

class SignInForm extends Component {
  render() {
    const { handleSubmit } = this.props;
    return (
      <div>
        <h2>New Person</h2>
        <form onSubmit={handleSubmit}>
          <Field name="firstName" component={ErrorField} />
          <Field name="lastName" component={ErrorField} />
          <Field name="email" component={ErrorField} />
          <div>
            <input type="submit" />
          </div>
        </form>
      </div>
    );
  }
}

SignInForm.propTypes = {};
SignInForm.defaultProps = {};

const validate = ({ firstName, lastName, email }) => {
  const errors = {};

  if (!firstName) {
    errors.firstName = 'firstName is required'
  }

  if (!lastName) {
    errors.lastName = 'lastName is required'
  }

  if (!email) {
    errors.email = 'email is required'
  } else if (!emailValidator.validate(email)) {
    errors.email = 'invalid email'
  }

  return errors
};

export default reduxForm({
  form: 'person',
  validate,
})(SignInForm);
