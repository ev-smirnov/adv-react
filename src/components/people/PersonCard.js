import React, { Component } from 'react';
import { DragSource } from 'react-dnd';

class PersonCard extends Component {
  render() {
    const { person, style, connectDragSource, isDragging } = this.props;
    const dragStyle = {
      backgroundColor: isDragging ? 'gray' : 'white',
    };
    return (
      <div style={{ width: 200, height: 100, ...dragStyle, ...style }}>
        {connectDragSource(<h3>{person.firstName}&nbsp;{person.lastName}</h3>)}
        <p>{person.email}</p>
      </div>
    )
  }
}

PersonCard.propTypes = {};
PersonCard.defaultProps = {};

const spec = {
  beginDrag(props) {
    return {
      uid: props.person.uid,
    }
  },
  endDrag(props, monitor) {
    const personUid = props.person.uid;
    const dropResult = monitor.getDropResult();
    const eventUid = dropResult && dropResult.uid;

    console.log('--- endDrag', personUid, eventUid);
  },
};

const collect = (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging(),
});

export default DragSource('person', spec, collect)(PersonCard);
