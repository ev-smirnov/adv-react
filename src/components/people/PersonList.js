import React, { Component } from 'react';
import { Table, Column } from 'react-virtualized';
import Loader from '../common/Loader';

class PersonList extends Component {
  componentDidMount() {
    this.props.fetchAll && this.props.fetchAll();
  }

  rowGetter = ({ index }) => this.props.people[index];

  render() {
    const { people, loading } = this.props;

    if (loading) {
      return <Loader />;
    }

    return (
      <Table
        rowCount={people.length}
        rowGetter={this.rowGetter}
        rowHeight={40}
        headerHeight={50}
        width={600}
        height={300}
        rowClassName="test--people-list__row"
      >
        <Column
          label="First Name"
          dataKey="firstName"
          width={200}
        />
        <Column
          label="Last Name"
          dataKey="lastName"
          width={200}
        />
        <Column
          label="Email"
          dataKey="email"
          width={200}
        />
      </Table>
    );
  }
}

PersonList.propTypes = {};
PersonList.defaultProps = {};

export default PersonList;
