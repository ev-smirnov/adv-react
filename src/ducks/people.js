import { Record, List } from 'immutable';
import { reset } from 'redux-form';
import { appName } from '../config';
import { put, call, takeEvery } from 'redux-saga/effects';
import { generateId } from './utils';

/**
 * Constants
 * */
export const moduleName = 'people';
const prefix = `${appName}/${moduleName}`;

export const ADD_PERSON_REQUEST = `${prefix}/ADD_PERSON_REQUEST`;
export const ADD_PERSON = `${prefix}/ADD_PERSON`;

/**
 * Reducer
 * */
const ReducerState = Record({
  entities: new List([]),
});

const PersonRecord = Record({
  id: null,
  firstName: null,
  lastName: null,
  email: null,
});

export default (state = new ReducerState(), action) => {
  const { type, payload } = action;

  switch (type) {
    case ADD_PERSON:
      return state.update('entities', entities => entities.push(new PersonRecord(payload)));

    default:
      return state
  }
};

/**
 * Selectors
 * */

/**
 * Action Creators
 * */
export const addPerson = (person) => ({
  type: ADD_PERSON_REQUEST,
  payload: person
});

/**
 * Sagas
 * */
export const addPersonSaga = function* (action) {
  const id = yield call(generateId);
  yield put({
    type: ADD_PERSON,
    payload: { ...action.payload, id }
  });

  yield put(reset('person'));
};

export const saga = function* () {
  yield takeEvery(ADD_PERSON_REQUEST, addPersonSaga);
};
