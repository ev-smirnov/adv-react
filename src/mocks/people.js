export const personDataModel = {
  firstName: 'name.firstName',
  lastName: 'name.lastName',
  email: 'internet.email',
};

export const personUidDataModel = {
  uid: (fixtures) => Math.random().toString(),
};

export const savedPersonDataModel = {
  ...personUidDataModel,
  ...personUidDataModel,
};

export const testPeople = [
  {
    "firstName": "Maurine",
    "lastName": "Nikolaus",
    "email": "Joel_Runte@gmail.com",
    "uid": "0.7585421471512288"
  },
  {
    "firstName": "Alec",
    "lastName": "Ortiz",
    "email": "Aida_Graham@hotmail.com",
    "uid": "0.21225517381780312"
  },
  {
    "firstName": "Fatima",
    "lastName": "Osinski",
    "email": "Jacques.Dibbert75@hotmail.com",
    "uid": "0.621659885609134"
  },
  {
    "firstName": "Madison",
    "lastName": "Bogan",
    "email": "Winfield_Ledner@gmail.com",
    "uid": "0.9857900821039924"
  },
  {
    "firstName": "Haleigh",
    "lastName": "Kautzer",
    "email": "Jovani2@gmail.com",
    "uid": "0.5439441864672319"
  },
  {
    "firstName": "Karlee",
    "lastName": "Fay",
    "email": "Bobbie24@gmail.com",
    "uid": "0.1521983696857745"
  },
  {
    "firstName": "Elmira",
    "lastName": "Schamberger",
    "email": "Rosemary_Watsica@yahoo.com",
    "uid": "0.8180847237740319"
  },
  {
    "firstName": "Kameron",
    "lastName": "Hudson",
    "email": "Louisa81@hotmail.com",
    "uid": "0.6090714869100333"
  },
  {
    "firstName": "Rudy",
    "lastName": "Bradtke",
    "email": "Davon.Dibbert@gmail.com",
    "uid": "0.6366397422250882"
  },
  {
    "firstName": "Vance",
    "lastName": "Kessler",
    "email": "Kathryn_Nader90@gmail.com",
    "uid": "0.5102227376521873"
  },
  {
    "firstName": "Dayne",
    "lastName": "Osinski",
    "email": "Jeffrey98@yahoo.com",
    "uid": "0.8923661699212371"
  },
  {
    "firstName": "Delores",
    "lastName": "Swaniawski",
    "email": "Adrien_Wiegand26@gmail.com",
    "uid": "0.1354868701699865"
  },
  {
    "firstName": "Imelda",
    "lastName": "Effertz",
    "email": "Kevon_Batz@hotmail.com",
    "uid": "0.0631360918045969"
  },
  {
    "firstName": "Adolph",
    "lastName": "Brekke",
    "email": "Heather_Bosco55@gmail.com",
    "uid": "0.27407315518263986"
  },
  {
    "firstName": "Edwina",
    "lastName": "Dicki",
    "email": "Monte1@gmail.com",
    "uid": "0.7288138732293716"
  },
  {
    "firstName": "Aracely",
    "lastName": "Barrows",
    "email": "Anastacio_Morissette52@hotmail.com",
    "uid": "0.21779903621308683"
  },
  {
    "firstName": "Percival",
    "lastName": "Padberg",
    "email": "Liana.Jones38@yahoo.com",
    "uid": "0.8482889089022414"
  },
  {
    "firstName": "Mayra",
    "lastName": "Bogan",
    "email": "Meagan.Hilpert11@hotmail.com",
    "uid": "0.9103326072266451"
  },
  {
    "firstName": "Eve",
    "lastName": "Wiegand",
    "email": "Janessa_Buckridge37@hotmail.com",
    "uid": "0.808123499149475"
  },
  {
    "firstName": "Scotty",
    "lastName": "Donnelly",
    "email": "Arianna32@gmail.com",
    "uid": "0.05876263880834487"
  }
];