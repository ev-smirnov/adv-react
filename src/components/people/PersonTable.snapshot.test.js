import React from 'react';
import PersonTable from './PersonTable';
import renderer from 'react-test-renderer';
import { testPeople } from '../../mocks/people';

jest.mock('react-dom', () => ({
  findDOMNode: () => ({}),
}));

it('should renders correctly', () => {
  const tree = renderer.create(<PersonTable people={testPeople} />);
  expect(tree).toMatchSnapshot();
});
