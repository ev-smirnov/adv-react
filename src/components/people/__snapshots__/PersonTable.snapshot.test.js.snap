// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<div
  className="ReactVirtualized__Table"
  id={undefined}
  role="grid"
  style={Object {}}
>
  <div
    className="ReactVirtualized__Table__headerRow test--people-list__row"
    role="row"
    style={
      Object {
        "height": 50,
        "overflow": "hidden",
        "paddingRight": 0,
        "width": 600,
      }
    }
  >
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 200px",
          "flex": "0 1 200px",
          "msFlex": "0 1 200px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="First Name"
      >
        First Name
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 200px",
          "flex": "0 1 200px",
          "msFlex": "0 1 200px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Last Name"
      >
        Last Name
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 200px",
          "flex": "0 1 200px",
          "msFlex": "0 1 200px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Email"
      >
        Email
      </span>
    </div>
  </div>
  <div
    aria-label="grid"
    className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
    id={undefined}
    onScroll={[Function]}
    role="rowgroup"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "boxSizing": "border-box",
        "direction": "ltr",
        "height": 250,
        "overflowX": "hidden",
        "overflowY": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
    tabIndex={0}
  >
    <div
      className="ReactVirtualized__Grid__innerScrollContainer"
      style={
        Object {
          "height": 800,
          "maxHeight": 800,
          "maxWidth": 600,
          "overflow": "hidden",
          "pointerEvents": "",
          "position": "relative",
          "width": "auto",
        }
      }
    >
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 0,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Maurine"
        >
          Maurine
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Nikolaus"
        >
          Nikolaus
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Joel_Runte@gmail.com"
        >
          Joel_Runte@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 40,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Alec"
        >
          Alec
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Ortiz"
        >
          Ortiz
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Aida_Graham@hotmail.com"
        >
          Aida_Graham@hotmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 80,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Fatima"
        >
          Fatima
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Osinski"
        >
          Osinski
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Jacques.Dibbert75@hotmail.com"
        >
          Jacques.Dibbert75@hotmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 120,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Madison"
        >
          Madison
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Bogan"
        >
          Bogan
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Winfield_Ledner@gmail.com"
        >
          Winfield_Ledner@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 160,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Haleigh"
        >
          Haleigh
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Kautzer"
        >
          Kautzer
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Jovani2@gmail.com"
        >
          Jovani2@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 200,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Karlee"
        >
          Karlee
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Fay"
        >
          Fay
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Bobbie24@gmail.com"
        >
          Bobbie24@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 240,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Elmira"
        >
          Elmira
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Schamberger"
        >
          Schamberger
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Rosemary_Watsica@yahoo.com"
        >
          Rosemary_Watsica@yahoo.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 280,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Kameron"
        >
          Kameron
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Hudson"
        >
          Hudson
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Louisa81@hotmail.com"
        >
          Louisa81@hotmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 320,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Rudy"
        >
          Rudy
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Bradtke"
        >
          Bradtke
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Davon.Dibbert@gmail.com"
        >
          Davon.Dibbert@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 360,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Vance"
        >
          Vance
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Kessler"
        >
          Kessler
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Kathryn_Nader90@gmail.com"
        >
          Kathryn_Nader90@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 400,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Dayne"
        >
          Dayne
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Osinski"
        >
          Osinski
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Jeffrey98@yahoo.com"
        >
          Jeffrey98@yahoo.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 440,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Delores"
        >
          Delores
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Swaniawski"
        >
          Swaniawski
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Adrien_Wiegand26@gmail.com"
        >
          Adrien_Wiegand26@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 480,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Imelda"
        >
          Imelda
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Effertz"
        >
          Effertz
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Kevon_Batz@hotmail.com"
        >
          Kevon_Batz@hotmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 520,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Adolph"
        >
          Adolph
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Brekke"
        >
          Brekke
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Heather_Bosco55@gmail.com"
        >
          Heather_Bosco55@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 560,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Edwina"
        >
          Edwina
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Dicki"
        >
          Dicki
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Monte1@gmail.com"
        >
          Monte1@gmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 600,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Aracely"
        >
          Aracely
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Barrows"
        >
          Barrows
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Anastacio_Morissette52@hotmail.com"
        >
          Anastacio_Morissette52@hotmail.com
        </div>
      </div>
      <div
        className="ReactVirtualized__Table__row test--people-list__row"
        role="row"
        style={
          Object {
            "height": 40,
            "left": 0,
            "overflow": "hidden",
            "paddingRight": 0,
            "position": "absolute",
            "top": 640,
            "width": 600,
          }
        }
      >
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Percival"
        >
          Percival
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Padberg"
        >
          Padberg
        </div>
        <div
          className="ReactVirtualized__Table__rowColumn"
          role="gridcell"
          style={
            Object {
              "WebkitFlex": "0 1 200px",
              "flex": "0 1 200px",
              "msFlex": "0 1 200px",
              "overflow": "hidden",
            }
          }
          title="Liana.Jones38@yahoo.com"
        >
          Liana.Jones38@yahoo.com
        </div>
      </div>
    </div>
  </div>
</div>
`;
